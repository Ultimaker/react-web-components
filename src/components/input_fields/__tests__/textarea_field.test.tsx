// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import TextareaField, {TextareaFieldProps} from '../textarea_field';
import Textarea from '../../textarea';
import InputFieldWrapper from '../input_field_wrapper';

describe('The textarea field component', () => {
    let props: TextareaFieldProps;
    let wrapper;

    beforeEach(() => {
        props = {
            id: 'testInputField',
            value: "2018",
            onChangeHandler: jest.fn(),
            labelLayoutWidth: '1/1',
            labelWidthBreakpoint: 'sm',
            staticField: false,
            placeholder: "placeholder textarea",
        };
        wrapper = shallow(<TextareaField {...props} />);
    });

    it('should render a wrapper', () => {
        expect(wrapper).toMatchSnapshot();
        expect(props.onChangeHandler).not.toHaveBeenCalled();
    });

    it('should render a null', () => {
        wrapper.setProps({value: null})
        expect(wrapper.find(Textarea).prop("value")).toBeNull()
    })

    it('should render a wrapped textarea', () => {
        expect(wrapper.find(Textarea).props()).toEqual({
            id: props.id,
            onChangeHandler: wrapper.instance()._onChange,
            placeholder: props.placeholder,
            value: props.value,
            autofocus: props.focusOnLoad,
            autoGrow: props.textareaAutoGrow,
        });
    })

    it('should render a static textarea', () => {
        wrapper.setProps({staticField: true})
        expect(wrapper.find(InputFieldWrapper).prop("children").map(c => c.props.children)).toEqual([props.value]);
    })

    it('should call the callback', () => {
        expect(props.onChangeHandler).not.toHaveBeenCalled();
        wrapper.find(Textarea).prop("onChangeHandler")("some text")
        expect(props.onChangeHandler).toHaveBeenLastCalledWith(props.id, "some text");
        wrapper.find(Textarea).prop("onChangeHandler")("")
        expect(props.onChangeHandler).toHaveBeenLastCalledWith(props.id, "");
    });
});
