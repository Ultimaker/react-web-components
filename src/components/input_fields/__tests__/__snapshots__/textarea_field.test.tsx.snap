// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The textarea field component should render a wrapper 1`] = `
<InputFieldWrapper
  className="input-field--textarea"
  id="testInputField"
  labelLayoutWidth="1/1"
  labelWidthBreakpoint="sm"
  onChangeHandler={[MockFunction]}
  showValidationError={true}
  staticField={false}
  touched={false}
>
  <Textarea
    id="testInputField"
    onChangeHandler={[Function]}
    placeholder="placeholder textarea"
    value="2018"
  />
</InputFieldWrapper>
`;
